### PR TITLE
kafka-value-parser

### PR DESCRIPTION
**Pull request by sworduo**
issue https://github.com/vesoft-inc/nebula-exchange/issues/8
import data from kafka to nebula

### CODE REVIEW

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala [hunk 1]
```diff
@@ -25,17 +28,25 @@ abstract class StreamingBaseReader(override val session: SparkSession) extends R
   * @param session
   * @param kafkaConfig
   */
-class KafkaReader(override val session: SparkSession, kafkaConfig: KafkaSourceConfigEntry)
+class KafkaReader(override val session: SparkSession,
+                  kafkaConfig: KafkaSourceConfigEntry,
+                  fields: List[String])
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:33]: please distinct the fields before using it.

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala [hunk 2]
```diff
@@ -25,17 +28,25 @@ abstract class StreamingBaseReader(override val session: SparkSession) extends R
...
-      .option("subscribe", kafkaConfig.topic)
-      .load()
+           .format("kafka")
+           .option("kafka.bootstrap.servers", kafkaConfig.server)
+           .option("subscribe", kafkaConfig.topic)
+           .load()
+           .selectExpr("CAST(value AS STRING)")
+           .as[(String)]
+           .withColumn("value", from_json(col("value"), jsonSchema))
+           .select("value.*")
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:49]: don't we need alias the dataframe's column name to name in fields?

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/Exchange.scala [hunk 3]
```diff
@@ -172,7 +173,8 @@ object Exchange {
         LOG.info(s"field keys: ${fieldKeys.mkString(", ")}")
         val nebulaKeys = edgeConfig.nebulaFields
         LOG.info(s"nebula keys: ${nebulaKeys.mkString(", ")}")
-        val data = createDataSource(spark, edgeConfig.dataSourceConfigEntry)
+        val fields = edgeConfig.sourceField::edgeConfig.targetField::edgeConfig.fields
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/Exchange.scala:176]: edgeConfig.rankField should also be added.

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala (hunk 2 above)
Comment (sworduo) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:49]: I had ever print the col name which is the name of fields. It's workable in my machine..

File: nebula-exchange/src/test/resources/application.conf [hunk 4]
```diff
@@ -224,23 +224,23 @@
...
-      fields: [kafka-field-0, kafka-field-1, kafka-field-2]
-      nebula.fields: [nebula-field-0, nebula-field-1, nebula-field-2]
-      vertex: {
-        field: kafka-field-0
-      }
-      partition: 10
-      batch: 10
-      interval.seconds: 10
-    }
+    # {
```
Comment (Nicole00) [inline nebula-exchange/src/test/resources/application.conf:227]: please roll back these changes

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala (hunk 2 above)
Comment (Nicole00) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:50]: Yeah, i tested it and the schema is the same with fields. Great work~

File: nebula-exchange/src/test/resources/application.conf (hunk 4 above)
Comment (sworduo) [reply nebula-exchange/src/test/resources/application.conf:227]: A new check is added in the config parse that an exception would be throw if any other config define after kafka, see Config.scala. However, there is two kafka defined in  the application.conf. If I don't comment this section, the test would not  pass.

Comment (Nicole00) [reply nebula-exchange/src/test/resources/application.conf:227]: Ok， I'll split the config of Kafka out and use a single config file for Kafka later.

**Review by Nicole00 — APPROVED** [review]
Great PR ~

**Pull request merged**